test/apiserver_organization_test.go

```diff
@@ -153,3 +156,161 @@ modfor:
+	require.NoError(t, err)
+	cfg.UserAgent = t.Name()
+	cl := testutil.NewRecordingClient(t, cfg, testScheme, testutil.CleanUpStrategy(cleanUpStragety))
+	t.Cleanup(cl.CleanUpFunc(ctx))
+
+	owner := rbacv1.Subject{
+		Kind:     "User",
+		APIGroup: "rbac.authorization.k8s.io",
+		Name:     "kubernetes-admin",
+	}
+	testCase := []*struct {
+		Name    string
+		Subject rbacv1.Subject
+		Imp     rest.ImpersonationConfig
+		Org     *apiserverv1alpha1.Organization
+	}{
+		{
+			Name: "user",
+			Subject: rbacv1.Subject{
+				Kind:     rbacv1.UserKind,
+				APIGroup: rbacv1.GroupName,
+				Name:     "user",
+			},
+			Imp: rest.ImpersonationConfig{
+				UserName: "user",
+			},
+		},
+		{
+			Name: "group",
+			Subject: rbacv1.Subject{
+				Kind:     rbacv1.GroupKind,
+				APIGroup: rbacv1.GroupName,
+				Name:     "group",
+			},
+			Imp: rest.ImpersonationConfig{
+				UserName: "lala",
+				// without "system:authenticated" things are breaking...cannot RESTMapper doesn't function
+				Groups: []string{"system:authenticated", "group"},
+			},
+		},
+		{
+			Name: "sa",
+			Subject: rbacv1.Subject{
+				Kind:      rbacv1.ServiceAccountKind,
+				Name:      "default",
+				Namespace: "default",
+			},
+			Imp: rest.ImpersonationConfig{
+				UserName: "system:serviceaccount:default:default",
+			},
+		},
+	}
+
+	t.Log("creating orgs")
+	for _, tc := range testCase {
+		tc.Org = &apiserverv1alpha1.Organization{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "test-" + tc.Name,
+				Labels: map[string]string{
+					"test-name": t.Name(),
+				},
+			},
+			Spec: corev1alpha1.OrganizationSpec{
+				Metadata: &corev1alpha1.OrganizationMetadata{
+					DisplayName: "test",
+					Description: "desc",
+				},
+				Owners: []rbacv1.Subject{owner},
+			},
+		}
+		require.NoError(t, cl.Create(ctx, tc.Org))
+	}
+
+	t.Log("waiting for orgs ready")
+	for _, tc := range testCase {
+		require.NoError(t, testutil.WaitUntilReady(ctx, cl, tc.Org))
+	}
+
+	t.Log("creating rolebindings in the orgs")
+	for _, tc := range testCase {
+		rb := &rbacv1.RoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "rb",
+				Namespace: tc.Org.Status.Namespace.Name,
+			},
+			RoleRef: rbacv1.RoleRef{
+				APIGroup: rbacv1.GroupName,
+				Kind:     "Role",
+				Name:     "test",
+			},
+			Subjects: []rbacv1.Subject{tc.Subject},
+		}
+		require.NoError(t, cl.Create(ctx, rb))
+	}
+
+	t.Log("creating necessary cluster roles/rolebinding")
+	crole := &rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "bulward:test-visible-filtering",
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				Verbs:     []string{rbacv1.VerbAll},
+				APIGroups: []string{apiserverv1alpha1.SchemeGroupVersion.Group},
+				Resources: []string{rbacv1.ResourceAll},
+			},
+		},
+	}
+	crolebinding := &rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "bulward:test-visible-filtering",
+		},
+		RoleRef: rbacv1.RoleRef{
+			Name: crole.Name,
+			Kind: "ClusterRole",
+		},
+	}
```

Comment:
This crole and crolebinding will not be needed once #31 gets merged

Reply:
These Role and RoleBindings are not needed, Owner has permission to create role/rolebindings since #31 is merged.

Reply:
they are needed, will reply in other post why

Reply:
They are needed, the users aren't owners, just an ordinary user of the organization created via the role. And how will you create an organization in the first place if you lack permissions doing so?

Reply:
In my opinion, we should guarantee both "users" and "owners" permission to manage Organizations. For example:
1. Every bulward user should have permission to create Organization (Then the user becomes an Owner of the Organization that just created).
2. User can only delete owned Organizations. This is extra important because currently, if we guarantee full permission for users in bulward to interact with Organizations, the user can also delete the Organization which is not owned by the user (From the `isVisable` method, we didn't check if the user is owner, so in the `Delete` method, user can deleted any Organization that the user is in.)

I think the easiest way to solve this issue is that we just give full permissions to manage Organizations to bulward, but we handle extra filtering in our apiserver, for example, for deletion, we check if the user is Owner of the organization, we also can do extra filtering in `Update` (for example, only Owner can update or whatever). WDYT?